### DISCUSSION TITLE
KolButton mit Loading Spinner

### DISCUSSION THREAD

**Discussion by Chrisdo82** [public-ui/kolibri#2586] [Q&A]
Hi,

wenn ich in einem Button einen Spinner mit anzeige, frage ich mich gerade welche zusätzlichen Attribute ich noch mitgeben muss, damit auch wirklich alle Benutzer abgeholt werden. 

Wenn ich z.B. bei einem Speichern Button 
![Bildschirm­foto 2023-03-08 um 13 34 05](https://user-images.githubusercontent.com/6606913/223714677-fab45554-70c9-4f5f-b2f9-62018d34d901.png)
beim Klick einen Spinner anzeige und den Text von `Speichern` zu `Wird Gespeichert` ändere
![Bildschirm­foto 2023-03-08 um 13 34 11](https://user-images.githubusercontent.com/6606913/223714876-661d2d0e-ebb5-4866-8c71-52c774dccceb.png)
kann ich dann 

1. den Button auch auf `_disabled` setzen oder sollte ich es besser im `onClick`-Handler abfangen dass der Button kein weiteres mal betätigt werden kann?
2. dem Button `aria-busy=true` mitgeben, damit Screenreadernutzer auch darüber informiert werden?

Gibt es noch weitere Sachen die dabei beachtet werden müssen?

---

Mir ist übrigens aufgefallen, dass die Kombination `_customClass="loading"`, `_variant="custom"` und `_icon="fas fa-spinner"` nicht mehr benötigt wird. Ein einfaches `_icon="fas fa-spinner fa-spin` tut es jetzt auch. Früher war das nicht möglich.

**Comment by deleonio** [comment]
Hallo @Chrisdo82 ,

wir wollten diese Feature "busy" in der Komponente umsetzen. Das ist im Issue https://github.com/public-ui/kolibri/issues/2476 platziert.

Ein fertiges Konzept gibt es noch nicht, aber ggf. könnten wir den Button / Link um `_aria-busy` erweitern, das disabled automatisch und macht den Rest. Man müsste dann aber den Style trotzdem losgelöst davon setzen.

**Reply by Chrisdo82** [reply]
Eigentlich wollte ich nur wissen was ich bei uns einbauen bzw. dabei beachten muss, damit es barrierefrei ist.

**Reply by Chrisdo82** [reply]
Nicht mehr relevant, kann geschlossen werden @deleonio
